cmd/kube-proxy/app/server.go

```diff
@@ -469,26 +470,25 @@ func NewProxyCommand() *cobra.Command {
 
 	cmd := &cobra.Command{
 		Use: "kube-proxy",
-		Long: `The Kubernetes network proxy runs on each node. This
-reflects services as defined in the Kubernetes API on each node and can do simple
-TCP, UDP, and SCTP stream forwarding or round robin TCP, UDP, and SCTP forwarding across a set of backends.
-Service cluster IPs and ports are currently found through Docker-links-compatible
-environment variables specifying ports opened by the service proxy. There is an optional
-addon that provides cluster DNS for these cluster IPs. The user must create a service
-with the apiserver API to configure the proxy.`,
+		Long: `The Kubernetes network proxy generally should run on each node of a cluster which needs to access Service IPs. It
+reflects services (as defined in the Kubernetes API) on each node and can do simple
+TCP, UDP, and SCTP stream forwarding or round robin TCP, UDP, and SCTP forwarding to backend IP addresses, or "endpoints".
+For DNS, consult your node's DNS implementation, which usually is aware of the "kube-dns" internal service (which maps services to virtual IP addresses, which are routed via kube-proxy).
+If pods/nodes don't have access to dynamic DNS, then service IPs are redundantly defined in environment variables available in the "docker links" format (SERVICE_PORT_PROTOCOL, i.e. DB_PORT_5432_TCP) for pre-existing services.
+These variables are are injected into a pod on startup.`,
```

Comment:
I feel like we shouldn't even mention the environment variables here:
1. There are important details that we don't have enough room to explain here (eg, it only covers services in the same namespace; it only covers services created before the pod itself, etc)
2. The environment variables aren't created by kube-proxy, they're created by kubelet, so why do we document them here?

I think the fact that they're documented in the full kubernetes documentation is enough. No one is going to expect to be able to learn everything about kubernetes services just by running `kube-proxy --help`.